### PR TITLE
Task-49264: Space members shouldn't be able to create projects

### PR DESCRIPTION
**Pull request by AzmiTouil**
Restrict project creation in space for space managers only.

### CODE REVIEW

File: services/src/main/java/org/exoplatform/task/rest/ProjectRestService.java [hunk 1]
```diff
@@ -760,6 +764,37 @@ public Response getProjectParticipants(@ApiParam(value = "Project id", required
     }
   }
 
+  @GET
+  @Path("canCreateProject/{spaceId}")
+  @Produces(MediaType.TEXT_PLAIN)
+  @RolesAllowed("users")
+  @ApiOperation(value = "check if the current user can create a project in the given space", httpMethod = "GET", response = Response.class, notes = "This checks if the current user can create a project in the given space", consumes = "application/json")
```
Comment (ahamdi) [inline services/src/main/java/org/exoplatform/task/rest/ProjectRestService.java]: Could you check if there is already an endpoint on Social side for checking this ?

File: task-management/src/main/webapp/vue-app/tasks-management/components/Project/ProjectListToolbar.vue [hunk 2]
```diff
@@ -93,12 +94,24 @@ export default {
...
+  },
   mounted() {
     $('#widthTmpSelectOption').html($('#filterTaskSelect option:selected').text());
     const widthElem = $('#widthTmpSelectTaskFilter').width() + 40;
     $('#filterTaskSelect').width(widthElem);
   },
   methods: {
+    canCreateProject() {
+      if (eXo.env.portal.spaceId) {
+        return this.$projectService.canCreateProject().then(canCreateProject => {
```
Comment (ahamdi) [inline task-management/src/main/webapp/vue-app/tasks-management/components/Project/ProjectListToolbar.vue]: Check if there is not already a service that returns the same info

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/exoplatform/task/rest/ProjectRestService.java (hunk 1 above)
Comment (AzmiTouil) [reply services/src/main/java/org/exoplatform/task/rest/ProjectRestService.java]: checked, there is no api for this.
Do you propose to add one in social API?

Comment (ahamdi) [reply services/src/main/java/org/exoplatform/task/rest/ProjectRestService.java]: Yep a generic one, to check if a user is a manger/super manager of a space, we need to return the list of space managers of a space in org/exoplatform/social/rest/api/SpaceRestResources.java and implementation in org/exoplatform/social/rest/impl/space/SpaceRestResourcesV1.java

Comment (ahamdi) [reply services/src/main/java/org/exoplatform/task/rest/ProjectRestService.java]: @AzmiTouil please make sure that the new endpoint is well secured, just space members and managers/supermanagers are able to access such endpoint

Comment (ahamdi) [reply services/src/main/java/org/exoplatform/task/rest/ProjectRestService.java]: @AzmiTouil it seems there is already an endpoint. But I think that we need to improve and retrieve that information from the space DTO object. 
We initialize the task components based on different level of memberships that we receive from the Space DTO object

**Review by ahamdi — APPROVED** [review]

**Pull request merged**